components/script/script_runtime.rs

```diff
@@ -581,9 +615,15 @@ impl Runtime {
         InitConsumeStreamCallback(cx, Some(consume_stream), Some(report_stream_error));
 
         let microtask_queue = Rc::new(MicrotaskQueue::default());
+        let mut interrupt_microtask_queues: Rc<Vec<Rc<MicrotaskQueue>>> = Rc::new(vec![]);
+        // TODO: SAFETY: is this ok? It subverts the Rc::get_mut() check
+        let interrupt_queues_raw = Rc::get_mut(&mut interrupt_microtask_queues)
```

Comment:
This feels like a subversion of the type system. If we're going to use Rc and mutate the contents then we should use `Rc<RefCell<...>>` and avoid this sketchy get_mut dance. Alternatively, if we actually believe we're modelling unique ownership, we should use a Box instead and do the forget/from_raw pattern in the constructor and destructor.